### DISCUSSION TITLE
Some shows have their Trakt ID appended on the end of their Slug

### DISCUSSION THREAD

**Discussion by mpfc75** [trakt/trakt-api#285] [Q&A] [answered]
I am noticing that the Slug for some shows has the Trakt ID appended to the end of the Slug. Why is this happening?  Could it have anything to do with the move away from TVDB?  For example:

- Rebel (traktid=167157) - slugs rebel-2021 and rebel-2021-167157.

**Comment by rectifyer** [comment, marked as answer]
What specific API methods are giving different slugs for that show? Slugs need to be unique, so most likely it uses the ID version at first then was able to automatically reclaim the `rebel-2021` slug later on. We keep both so any references continue to work.

**Comment by mpfc75** [comment]
URLs https://trakt.tv/shows/rebel-2021-167157 and https://trakt.tv/shows/rebel-2021 both resolve to https://trakt.tv/shows/rebel-2021 and both have the slug rebel-2021.  

I noticed the existence of shows with the Trakt ID on the end because my app stores a local copy of the data using a key that starts with the 'slug'.  At some point the show 'rebel-2021-167157' was added to the watchlist (with a key ending in with the Trakt ID) but subsequently when the app was using ids.slug to generate a key for a search of season or episode (inside my app) it would generate various errors because the key generated would start with 'rebel-2021' (without the Trakt ID).

I am only raising this because I didn't know if it was the way it always worked (and I've just been lucky to have never encountered it before) or if it is part of the move away from TvDB.  Another show with Trakt ID on the end of the slug is 'The Chase' from 2021.  But in this case the two URLs have different slugs but you can see that they really are describing the same show.

- https://api.trakt.tv/shows/the-chase-2021-179295
- https://api.trakt.tv/shows/the-chase-us-2021

**Reply by rectifyer** [reply]
Yes, this is how slugs have always worked. We keep a history of slugs so we don't break any URLs. Looks like that is what happened with the first example.

If we can't create a unique slug using the title and year alone, it will append the Trakt ID to make sure it is unique. It's always best to save and use the Trakt ID if possible.
